internal/cmd/controller/controllers/cluster/import.go

```diff
@@ -188,33 +188,11 @@ func (i *importHandler) deleteOldAgent(cluster *fleet.Cluster, kc kubernetes.Int
 		return err
 	}
 
-	deployment, err := kc.AppsV1().Deployments(namespace).Get(i.ctx, config.AgentConfigName, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return nil
-	} else if err != nil {
-		return err
-	}
```

Comment:
Should we keep this or some other sort of backwards compatibility to support older agent versions that used `Deployment`?

Reply:
Not sure. I find this part problematic. The agent will delete it's own deployment (via Helm) when the bundle is updated. So this is more like a fail-safe?